rotkehlchen/accounting/cost_basis/base.py

```diff
@@ -463,43 +507,13 @@ def calculate_spend_cost_basis(
                 event=acquisition_event,
                 taxable=taxable,
             ))
+            asset_events.used_acquisitions.append(acquisition_event)
+            asset_events.acquisitions.report_result(acquisition_event.remaining_amount)
             # and since this events is going to be removed, reduce its remaining to zero
             acquisition_event.remaining_amount = ZERO
 
-            # If the sell used up the last historical acquisition
-            if idx == len(asset_events.acquisitions) - 1:
-                stop_index = idx + 1
-
-        if len(asset_events.acquisitions) == 0:
```

Comment:
You are removing so much logic from here? Where did it go?

Reply:
`if len(asset_events.acquisitions) == 0:` went nowhere :smiley: since it was redundant (other code in this method handles this case properly)

And this logic went to BaseAcquisitionsOrder
```
asset_events.used_acquisitions.extend(
    asset_events.acquisitions[:stop_index],
)
del asset_events.acquisitions[:stop_index]
# and modify the amount of the buy where we stopped if there is one
if remaining_amount_from_last_buy != FVal('-1'):
    asset_events.acquisitions[0].remaining_amount = remaining_amount_from_last_buy  # noqa: E501
```